src/services/privateKeyUtils.ts

```diff
@@ -44,6 +94,29 @@ export async function decryptPrivateKey(privateKeyInfo: PrivateKeyInfo, mnemonic
 	throw new Error('Failed to decrypt private key')
 }
 
+/**
+ * Hack to allow singing the CSR with the RSA key we created for encryption
+ * (reason is that its not recommended for RSA to use the same key for both signing and encryption)
```

Comment:
Oh yeah, I remember that limitation now, scared me at the time. Nice workaround :+1: